File: sdk/python/a3s_box/options.py
"""Sandbox configuration options."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass
class MountSpec:
    """A host-to-guest mount specification."""

    host_path: str
    guest_path: str
    readonly: bool = False


@dataclass
class PortForward:
    """Port forwarding rule: expose a guest port on the host."""

    guest_port: int
    host_port: int = 0
    protocol: str = "tcp"


@dataclass
class WorkspaceConfig:
    """Persistent workspace configuration."""

    name: str
    guest_path: str = "/workspace"


@dataclass
class SandboxOptions:
    """Options for creating a new sandbox."""

    image: str = "alpine:latest"
    cpus: int = 1
    memory_mb: int = 256
    env: dict[str, str] = field(default_factory=dict)
    workdir: str | None = None
    mounts: list[MountSpec] = field(default_factory=list)
    network: bool = True
    tee: bool = False
    name: str | None = None
    port_forwards: list[PortForward] = field(default_factory=list)
    workspace: WorkspaceConfig | None = None

File: sdk/typescript/src/options.ts
/**
 * Sandbox configuration options.
 */

export interface MountSpec {
  hostPath: string;
  guestPath: string;
  readonly?: boolean;
}

export interface PortForward {
  guestPort: number;
  hostPort?: number;
  protocol?: string;
}

export interface WorkspaceConfig {
  name: string;
  guestPath?: string;
}

export interface SandboxOptions {
  image?: string;
  cpus?: number;
  memoryMb?: number;
  env?: Record<string, string>;
  workdir?: string;
  mounts?: MountSpec[];
  network?: boolean;
  tee?: boolean;
  name?: string;
  portForwards?: PortForward[];
  workspace?: WorkspaceConfig;
}

export const DEFAULT_OPTIONS: Required<
  Pick<SandboxOptions, "image" | "cpus" | "memoryMb" | "network" | "tee">
> = {
  image: "alpine:latest",
  cpus: 1,
  memoryMb: 256,
  network: true,
  tee: false,
};
